### PR TITLE
WIP configs for an alb and http proxy

### PR DESCRIPTION
**Pull request by rnagle**
What's included:

- Terraform configurations for an ALB -> Nginx proxy -> s3 bucket

Why: we're currently unable to use Cloudfront as it is not considered an in-scope service. Our best option, at the moment, is to use an ALB and simple http proxy, since we _can_ assign a TLS cert to the ALB.

How to use this:

```
cd ops/terraform/bluebutton.cms.gov
terraform init -backend-config=</path/to/backend.tfvars>
terraform plan -var-file=</path/to/terraform.tfvars>
```

TODO:

- [ ] Finish configuration of TLS for the ALB
- [ ] Remove http listener

NOTE: I've left a few things out of this PR since they may contain sensitive information:

- backend.tfvars
- terraform.tfvars
- nginx.conf

### CODE REVIEW

File: ops/terraform/bluebutton.cms.gov/main.tf [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+    bucket_name = "${var.bucket_name}"
+    nat_gw_cdir = "${var.nat_gw_cdir}"
+  }
+}
+
+data "template_file" "user_data" {
+  template = "${file("${path.module}/templates/user_data")}"
+
+  vars {
+    bucket_name   = "${var.bucket_name}.s3-website-us-east-1.amazonaws.com"
```
Comment (whytheplatypus) [inline ops/terraform/bluebutton.cms.gov/main.tf]: looks like this isn't used by the template atm

File: ops/terraform/bluebutton.cms.gov/main.tf [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+    "${aws_security_group.instance_sg.id}",
+    "${data.aws_security_group.vpn.id}"
+  ]
+
+  key_name                    = "${var.key_name}"
+  image_id                    = "${var.ami_id}"
+  instance_type               = "${var.instance_type}"
+  user_data                   = "${data.template_file.user_data.rendered}"
+  associate_public_ip_address = false
+  iam_instance_profile        = "${var.instance_profile}"
```
Comment (whytheplatypus) [inline ops/terraform/bluebutton.cms.gov/main.tf]: is this something that can be built within terraform?

File: ops/terraform/bluebutton.cms.gov/main.tf (hunk 1 above)
Comment (rnagle) [reply ops/terraform/bluebutton.cms.gov/main.tf]: Good catch!

File: ops/terraform/bluebutton.cms.gov/main.tf (hunk 2 above)
Comment (rnagle) [reply ops/terraform/bluebutton.cms.gov/main.tf]: Yep, just being a little lax about including everything here. Mostly concerned with ability to launch and update the proxy server.

Comment (whytheplatypus) [reply ops/terraform/bluebutton.cms.gov/main.tf]: cool, just wondering

**Pull request merged**